src/runner/errors.ts

```diff
@@ -112,16 +113,17 @@ export async function toSourceError(error: Error, sourceMap?: RawSourceMap): Pro
     line = originalPosition.line ?? -1 // use -1 in place of null
     column = originalPosition.column ?? -1
     identifier = originalPosition.name ?? identifier
+    source = originalPosition.source ?? null
```

Comment:
Note that this will currently always return `source` as the value because it is hardcoded as the file name for the `SourceMapGenerator`:
https://github.com/source-academy/js-slang/blob/cb0b853caedfbde72a9674f1353af8cbff2bcfbd/src/transpiler/transpiler.ts#L619
https://github.com/source-academy/js-slang/blob/cb0b853caedfbde72a9674f1353af8cbff2bcfbd/src/transpiler/transpiler.ts#L650

However, if everything is working well, we should not encounter any native errors.